app/models/job_invocation.rb

```diff
@@ -204,17 +204,12 @@ def output(host)
   def generate_description
     key_re = /%\{([^\}]+)\}/
     template_invocation = pattern_template_invocations.first
-    hash_base = Hash.new { |hash, key| hash[key] = "%{#{key}}" }
-    input_hash = template_invocation.input_values.reduce(hash_base) do |h, v|
+    input_hash = template_invocation.input_values.reduce({}) do |h, v|
       h.update(v.template_input.name => v.value)
     end
-    input_hash.update(:job_category => job_category)
-    input_hash.update(:template_name => template_invocation.template.name)
-    description_format.scan(key_re) { |key| input_hash[key.first] }
-    self.description = description_format
-    input_hash.each do |k, v|
-      self.description.gsub!(Regexp.new("%\{#{k}\}"), v || '')
-    end
+    input_hash.update("job_category" => job_category)
+    input_hash.update("template_name" => template_invocation.template.name)
+    self.description = description_format.gsub(/%{([^}]+)}/) { |key| input_hash.fetch(Regexp.last_match[1], "''") }
```

Comment:
I just found out a cool feature - [`Hash#default=`](https://ruby-doc.org/core-2.5.1/Hash.html#method-i-default-3D), so we can do:
```suggestion
    input_hash.default = "''"
    self.description = description_format.gsub(/%{([^}]+)}/) { |key| input_hash[Regexp.last_match[1]] }
```